### PR TITLE
6 audio lattice

### LINKED ISSUE #6: Audio lattice type for LatticeTokenFilter
- [x] Be able to configure `LatticetokenFilter` token filter to be an 'audio' lattice or a 'lattice' lattice (for text)
- [x] Store the same information in a `MapperFieldType`
- [x] Have `LatticePayloadScoreQuery` that information.
  - [x] Should be able to configure `position_slop` for 'lattice' lattices and `time_slop` for 'audio' lattices
  - [x] Fix span width score normalization to make sense with the augmented position increments above.

### PR DESCRIPTION
**Pull request by messiaen**
fixes #6

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,10 +4,11 @@ MAKEFLAGS += --no-builtin-variables
 
 ES_VERSION = $(shell grep ES_VERSION .env | cut -d '=' -f 2)
 VERSION = $(shell grep PLUGIN_VERSION .env | cut -d '=' -f 2)
+COMMIT_HASH = $(shell grep COMMIT_HASH .env | cut -d '=' -f 2)
```
Comment (mjwall) [inline Makefile]: Maybe something like 
```
git describe --match=NeVeRmAtCh --always --abbrev=8 --dirty
```

File: src/main/java/com/eigendomain/eslatticeindex/index/LatticeWhitespaceTokenizer.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.apache.lucene.util.AttributeFactory;
+
+/**
+ * A tokenizer that divides text at whitespace characters as defined by
+ * {@link Character#isWhitespace(int)}.  Note: That definition explicitly excludes the non-breaking space.
+ * Adjacent sequences of non-Whitespace characters form tokens.
+ *
+ * @see UnicodeWhitespaceTokenizer
+ */
+public final class LatticeWhitespaceTokenizer extends CharTokenizer {
```
Comment (mjwall) [inline src/main/java/com/eigendomain/eslatticeindex/index/LatticeWhitespaceTokenizer.java]: I asked why this was needed, you said to work around the WhitespaceTokenizer not being loaded in tests.  Just making a note

File: Makefile [hunk 3]
```diff
@@ -26,6 +27,7 @@ push:
 test:
 	./gradlew cleanTest test
```
Comment (mjwall) [inline Makefile]: Not running `./gradlew integTest` anywhere.   Consider running tests before the assemble

File: Makefile [hunk 4]
```diff
@@ -4,10 +4,11 @@ MAKEFLAGS += --no-builtin-variables
 
 ES_VERSION = $(shell grep ES_VERSION .env | cut -d '=' -f 2)
 VERSION = $(shell grep PLUGIN_VERSION .env | cut -d '=' -f 2)
```
Comment (mjwall) [inline Makefile:6]: Maybe should bump the version in .env

**Review by mjwall — APPROVED** [review]

File: src/main/java/com/eigendomain/eslatticeindex/index/LatticeWhitespaceTokenizer.java (hunk 2 above)
Comment (messiaen) [reply src/main/java/com/eigendomain/eslatticeindex/index/LatticeWhitespaceTokenizer.java]: I've removed the custom class in favor of calling using the normal WhitespaceTokenizer in the custom factory.  This is still not great, but it a little better at least. Created #15 to fix this.

File: Makefile (hunk 1 above)
Comment (messiaen) [reply Makefile]: fixed

**Pull request merged**